src/Simplex/Messaging/Notifications/Server.hs

```diff
@@ -109,43 +110,71 @@ ntfSubscriber NtfSubscriber {smpSubscribers, newSubQ, smpAgent = ca@SMPClientAge
       forever $
         atomically (peekTBQueue subscriberSubQ) >>= \case
           NtfSub NtfSubData {smpQueue, notifierKey} -> do
+            updateSubStatus smpQueue NSPending
             let SMPQueueNtf {smpServer, notifierId} = smpQueue
             liftIO (runExceptT $ subscribeQueue ca smpServer ((SPNotifier, notifierId), notifierKey)) >>= \case
               Right _ -> do
-                -- update subscription status
+                updateSubStatus smpQueue NSActive
                 _ <- atomically $ readTBQueue subscriberSubQ
                 pure ()
               Left _e -> pure ()
 
     receiveSMP :: m ()
     receiveSMP = forever $ do
       (srv, _sessId, ntfId, msg) <- atomically $ readTBQueue msgQ
+      let smpQueue = SMPQueueNtf srv ntfId
       case msg of
         SMP.NMSG nmsgNonce encNMsgMeta -> do
           ntfTs <- liftIO getSystemTime
-          let smpQueue = SMPQueueNtf srv ntfId
           st <- asks store
           NtfPushServer {pushQ} <- asks pushServer
           atomically $
             findNtfSubscriptionToken st smpQueue
               >>= mapM_ (\tkn -> writeTBQueue pushQ (tkn, PNMessage PNMessageData {smpQueue, ntfTs, nmsgNonce, encNMsgMeta}))
+        SMP.END -> updateSubStatus smpQueue NSInactive
+        SMP.ERR AUTH -> updateSubStatus smpQueue NSSMPAuth
+        SMP.ERR e -> updateSubStatus smpQueue (NSSMPErr e)
         _ -> pure ()
       pure ()
 
     receiveAgent =
       forever $
         atomically (readTBQueue agentQ) >>= \case
           CAConnected _ -> pure ()
-          CADisconnected _srv _subs -> do
-            -- update subscription statuses
-            pure ()
+          CADisconnected srv subs ->
+            forM_ subs $ \(_, ntfId) -> do
+              let smpQueue = SMPQueueNtf srv ntfId
+              updateSubStatus smpQueue NSInactive
           CAReconnected _ -> pure ()
-          CAResubscribed _srv _sub -> do
-            -- update subscription status
-            pure ()
-          CASubError _srv _sub _err -> do
-            -- update subscription status
-            pure ()
+          CAResubscribed srv sub -> do
+            let ntfId = snd sub
+                smpQueue = SMPQueueNtf srv ntfId
+            updateSubStatus smpQueue NSActive
+          CASubError srv sub err -> do
+            let ntfId = snd sub
+                smpQueue = SMPQueueNtf srv ntfId
+            case err of
+              PCEProtocolError e -> case e of
+                AUTH -> updateSubStatus smpQueue NSSMPAuth
+                _ -> updateSubStatus smpQueue (NSSMPErr e)
+              PCEResponseError e -> updateSubStatus smpQueue (NSSMPErr e)
```

Comment:
```suggestion
              PCEResponseError e -> logErr err
```

Reply:
the reason is that we don't know what is the problem here, it's some response we couldn't parse - it's not the same as protocol error